tensorly/decomposition/_tucker.py

```diff
@@ -578,6 +578,12 @@ def non_negative_tucker_hals(
     if fixed_modes is None:
         fixed_modes = []
 
+    if tl.ndim(tensor) - 1 in fixed_modes:
+        warnings.warn(
+            "You asked for fixing the last mode, which is not supported.\n The last mode will not be fixed. Consider using tl.moveaxis()"
```

Comment:
```suggestion
            "You asked for fixing the last mode, which is not supported. The last mode will not be fixed."
            " Consider using tl.moveaxis() to permute it to another position and keep it fixed there."
```

Reply:
Perhaps we should do this ourselves in the future.

Reply:
We don't do it now because we use last factor for efficient computation. If you agree to transpose data inside the function when the last mode is fixed, I can change this PR.